src/services/ipaPublicAdministrationService.ts

```diff
@@ -24,31 +27,37 @@ export function upsertFromIpa(): Promise<void> {
       indicepaResponse.body
         .pipe(csvStream)
         .pipe(
-          es.map((entry: IIpaPublicAdministration, cb: () => void) => {
-            if (entry.cf_validato !== "S") {
-              // filter out entries without a validated CF
-              return cb();
-            }
-            if (!entry.Cf || !entry.Cf.match(/^\d{2,}$/)) {
-              // filter out entries with bogus CF
-              return cb();
-            }
-            const entryIndex = notUpdatedEntries.findIndex(
-              entryFromDb => entryFromDb.cod_amm === entry.cod_amm
-            );
-            notUpdatedEntries.splice(entryIndex, 1);
-            IpaPublicAdministrationModel.upsert(entry)
-              .then(() => {
-                cb();
-              })
-              .catch(error => {
-                log.error(
-                  "IpaPublicAdministration upsert failed for %s: %s",
-                  entry.des_amm,
-                  error
+          es.map((parsedRow: IIpaPublicAdministrationRaw, cb: () => void) => {
+            IpaPublicAdministration.decode(
+              Object.entries(parsedRow)
+                .map(([key, val]) => [key, val.trim()])
+                .reduce(
```

Comment:
consider also using `Object.fromEntries(Object.entries(...).map(...))` instead of reduce

Reply:
I also thought of this, but this method is supported since 12.0.0 of Node.js and [here](https://github.com/teamdigitale/io-onboarding-pa-api/blob/169034053-fix-public-administration-info-upsert-from-ipa/Dockerfile#L15) we use 10.14.1.